fdbserver/IPager.h

```diff
@@ -76,16 +77,69 @@ static const std::vector<std::pair<PagerEvents, PagerEventReasons>> L0PossibleEv
 	{ PagerEvents::PageWrite, PagerEventReasons::MetaData },
 };
 
-// Represents a block of memory in a 4096-byte aligned location held by an Arena.
+enum EncodingType : uint8_t {
+	XXHash64 = 0,
+	// For testing purposes
+	XOREncryption = 1
+};
+
+enum PageType : uint8_t {
+	HeaderPage = 0,
+	BackupHeaderPage = 1,
+	BTreeNode = 2,
+	BTreeSuperNode = 3,
+	QueuePageStandalone = 4,
+	QueuePageInExtent = 5
+};
+
+// Encryption key ID
+typedef uint64_t KeyID;
+
+// Encryption key and secret string
+struct KeyIDWithSecret {
+	KeyID id;
+	Key secret;
```

Comment:
a nit: reading "Key secret" to me was confusing initially, however, given the "Key" is defined as "StringRef" it perfectly makes sense. 
Do you think adding a new definition "typedef StringRef EncryptSecret" would make code bit more readable? (Just a thought)

Reply:
I did have the same thought but `Key` is actually `Standalone<StringRef>` and `KeyRef` is `StringRef` so `EncryptionKey/Ref` would need both of those.  It seemed a bit unnecessary since any reader of this code knows conceptually what a Key is, but I'm not against it.

It occurs to me now that perhaps EncryptionKey should perhaps be a `Ref` type with an optional key ID and replace the rename-pending EncryptionKeyInfo as EncryptionKeyInfoRef, with the non-Ref type being a typedef for Standalone (which is the usual fdb ref pattern).  And APIs should just use this struct for input/output instead of separating key ID and secret, because that basically implies / assumes that the only members of the struct are key and secret.  I started making this change and I like it.